Assets/Scripts/PlayerMovement.cs

```diff
@@ -64,13 +63,15 @@ public class PlayerMovement : MonoBehaviour {
         }
 
         if (jumping && grounded) {
-            grounded = false;
             rb.AddForce(new Vector2(0f, jump));
         }
 
-        if (dashing) {
+        if (dashing && canDash) {
             rb.AddForce(new Vector2(dash * (facingRight ? 1 : -1), 0f));
-            canDash = false;
+
+            if (!grounded) {
```

Comment:
If they aren't grounded, then they can only dash once and we set it to false. The loop above is what 'reset's their ability to dash. Later we can do something similar for dashing after breaking a wall/door thing